NDTensors/src/lib/Sectors/src/sector.jl

```diff
@@ -0,0 +1,128 @@
+
+struct Sector{Data} <: AbstractCategory
+  data::Data
+end
+
+Sector(nt::NamedTuple) = Sector{NamedTuple}(nt_sort(nt))
```

Comment:
Using `Sector{NamedTuple}` means the compiler doesn't have all of the type information about `data` anymore, so accessing `data` is type unstable and can hurt performance. Better to do something like:
```julia
function Sector(nt::NamedTuple)
  data = nt_sort(nt)
  return Sector{typeof(data)}(data)
end
```
or define an inner constructor `_Sector` that you call.